src/libs/actions/App.js

```diff
@@ -86,6 +86,7 @@ function setLocale(locale) {
     API.write('UpdatePreferredLocale', {
         value: locale,
     }, {optimisticData});
+    Navigation.navigate(ROUTES.SETTINGS_PREFERENCES);
```

Comment:
Hey that's what I recommended too :D 

https://github.com/Expensify/App/pull/14591#discussion_r1097051267

Reply:
I thought it's ok to put the navigation in App.setLocale since the other use case where this action is used returns early [here](https://github.com/Expensify/App/blob/c775563572bffc394c87616092f6291a616646fd/src/libs/actions/App.js#L74). But I do agree that it is more readable and clean to add a new method which clearly shows its purpose.